From 222d84a0de0d2dbd75b2c73f469d74868955f3b5 Mon Sep 17 00:00:00 2001
From: Guan Jing <guanjing6@huawei.com>
Date: Wed, 30 Nov 2022 18:23:25 +0800
Subject: [PATCH] sched/fair: Fix kabi borken in sched_domain_shared
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 222d84a0de0d2dbd75b2c73f469d74868955f3b5
Modified-by-SEL: No


hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I61E4M
CVE: NA

--------------------------------

The sched_domain_shared structure is only used as pointer, and other
drivers don't use it directly.

Signed-off-by: Guan Jing <guanjing6@huawei.com>
Reviewed-by: zhangjialin <zhangjialin11@huawei.com>
Reviewed-by: Zhang Qiao <zhangqiao22@huawei.com>
Reviewed-by: Chen Hui <judy.chenhui@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/sched/topology.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/sched/topology.h b/include/linux/sched/topology.h
index fe4e222d5d2d..249c98aef083 100644
--- a/include/linux/sched/topology.h
+++ b/include/linux/sched/topology.h
@@ -82,10 +82,10 @@ struct sched_domain_shared {
 	atomic_t	ref;
 	atomic_t	nr_busy_cpus;
 	int		has_idle_cores;
-	int		nr_idle_scan;
 #ifdef CONFIG_SCHED_STEAL
 	struct sparsemask *cfs_overload_cpus;
 #endif
+	KABI_EXTEND(int		nr_idle_scan)
 };
 
 struct sched_domain {
-- 
2.33.0

